### PR TITLE
Fixing the select with eq issue

### PR DESCRIPTION
**Pull request by ninaceban**
## Description
This is a draft PR for the issue opened by @paul-bjorkstrand. 

In At.js we are using Zepto to query DOM. In Web SDK we have partially transferred the functionality and have introduced a bug. Currently Web SDK doesn't correctly select children if `eq()` pseudo selector is involved. 
I have looked into Zepto library and I have followed the same approach.

## Related Issue
https://github.com/adobe/alloy/issues/850

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/utils/dom/querySelectorAll.js [hunk 1]
```diff
@@ -12,5 +12,22 @@ governing permissions and limitations under the License.
...
-export default (context, selector) =>
-  toArray(context.querySelectorAll(selector));
+const SIBLING_PATTERN = /^\s*>/;
+
+export default (context, selector) => {
+  if (!SIBLING_PATTERN.test(selector)) {
+    return toArray(context.querySelectorAll(selector));
+  }
+
+  const tag = `alloy${+new Date()}`;
```
Comment (jonsnyder) [inline src/utils/dom/querySelectorAll.js:22]: Although equivalent, it is more clear to change this to `alloy${new Date().getTime()}`.

File: src/utils/dom/querySelectorAll.js [hunk 2]
```diff
@@ -12,5 +12,22 @@ governing permissions and limitations under the License.
...
+
+  try { // adding a dummy css class to be able to select the children
```
Comment (jonsnyder) [inline src/utils/dom/querySelectorAll.js:24]: // We could use a :scope selector here, but we want to be IE compliant
// so we add a dummy css class to be able to select the children

File: src/utils/dom/querySelectorAll.js [hunk 3]
```diff
@@ -12,5 +12,22 @@ governing permissions and limitations under the License.
...
+    context.classList.add(tag); // needs to be IE compliant
```
Comment (jonsnyder) [inline src/utils/dom/querySelectorAll.js:25]: I'm not sure what you mean by this. From what I can tell classList.add/remove is supported in IE 10 and above.

**Review by jonsnyder — APPROVED** [review]

File: test/unit/specs/utils/dom/selectNodesWithShadow.spec.js [hunk 4]
```diff
@@ -136,4 +136,24 @@ describe("Utils::DOM::selectNodesWithShadow", () => {
...
+    const content = `
+      <div>
+        <div>
+          <span id="wrong"></span>
+        </div>
+        <span id="right"></span>
+      </div>
+    `;
+
+    const node = createNode("DIV", { id: "target" }, { innerHTML: content });
```
Comment (paul-bjorkstrand) [inline test/unit/specs/utils/dom/selectNodesWithShadow.spec.js:150]: I think this needs an `eq` class to allow the `afterEach` on [lines 75-77][1] to automatically clean up the markup that was added (so other tests function correctly).

Perhaps, this entire file of needs a refactor to make the purpose of the `eq` class more apparent (perhaps by renaming it to `auto-cleanup`, or change the `afterEach` to just empty the `body`, to ensure tests are executed in isolation?

[1]: https://github.com/paul-bjorkstrand/alloy/blob/main/test/unit/specs/components/Personalization/dom-actions/dom/selectNodesWithEq.spec.js#L75...L77

File: src/utils/dom/querySelectorAll.js (hunk 1 above)
Comment (ninaceban) [reply src/utils/dom/querySelectorAll.js]: @jonsnyder I have replaced it with `Date.now()` as far as I saw it is supported in IE 9+ and all other modern browsers.

File: src/utils/dom/querySelectorAll.js (hunk 3 above)
Comment (ninaceban) [reply src/utils/dom/querySelectorAll.js]: @jonsnyder it was a placeholder for me to double check IE compatibility. According to MDN it is supported in IE 10+ and modern browsers. I have removed the comments.

File: test/unit/specs/utils/dom/selectNodesWithShadow.spec.js (hunk 4 above)
Comment (ninaceban) [reply test/unit/specs/utils/dom/selectNodesWithShadow.spec.js]: @paul-bjorkstrand thanks for suggestion. Adjusted both `selectNodesWithEq.spec.js` and `selectNodesWithShadow.spec.js` to use `cleanup` CSS class to ensure `afterEach` get rids of all the elements that have been added during test run.

**Review by carterworks — APPROVED** [review]
Looks good to me. We can wait for @XDex to give the final word if you want.

**Pull request merged**